graphReconstruct/graphReconstruct.go

```diff
@@ -2,230 +2,91 @@ package graphReconstruct
-					} else if prevNode == node.Right {
-						sum = sum + Prob(i, j, node.Left.BranchLength)*Prob(i, k, node.Right.BranchLength)*scrap[k]*node.Left.Stored[j]
-					}
-				} else if prevNode.Up == nil {
-					sum = sum + Prob(i, j, node.Left.BranchLength)*Prob(i, k, node.Right.BranchLength)*node.Left.Stored[j]*node.Right.Stored[k]
-				}
+//BuildEdges connects the nodes of a species' graph that are stored in GraphColumns
+//func BuildEdges
+//start without prob
+//loop through species in column, go through all nodes
+//check if that node's seq matches the seq of the ancestor node without an edge
+//make the next of that node the same as this node's next
+//FindAncSeq creates a graph from the node records stored in GraphColumns and then calls PathFinder and seqOfPath to determine the most likley seq of the ancestor before assigning that
+//seq to the Fasta field of the ancestors tree node
+//func FindAncSeq will loop through aligncolumns and build a single graph of all of the nodes that belong to the ancestor species after edges are created
+//run PathFinder on the graph for the anc, run seqOfPath, then turn that to a fasta for that node of the tree
+
+//seqOfPath takes in a graph and a path specified by the Node IDs and returns the seq of the path through the graph
+func seqOfPath(g *simpleGraph.SimpleGraph, path []uint32) []dna.Base {
+	var seq []dna.Base
+	for n := 0; n < len(g.Nodes); n++ {
+		for p := 0; p < len(path); p++ {
+			if g.Nodes[n].Id == path[p] {
+				seq = append(seq, g.Nodes[n].Seq...)
 			}
 		}
-		scrapNew[i] = sum
-	}
-	if node.Up != nil {
-		BubbleUp(node.Up, node, scrapNew)
-	} else if node.Up == nil {
-		tot = scrapNew[0] + scrapNew[1] + scrapNew[2] + scrapNew[3]
-		node.Scrap = tot
 	}
+	return seq
 }
 
-//fix each node and return the probabilities for each base at that site
-func FixFc(root *expandedTree.ETree, node *expandedTree.ETree) []float64 {
-	ans := []float64{0, 0, 0, 0}
+//PathFinder takes a graph and returns the most likely path through that graph after checking all possible paths from the first node to the last
+func PathFinder(g *simpleGraph.SimpleGraph) ([]uint32, float32) {
+	var finalPath []uint32
+	var finalProb float32
+	var tempPath = make([]uint32, 0)
 
-	for i := 0; i < 4; i++ {
-		scrap := []float64{0, 0, 0, 0} //checking one base at a time each time you call BubbleUp
-		scrap[i] = node.Stored[i]
-		if node.Up != nil {
-			//Bubble up the tree using the memory of the previous node in relation to changing position taking in probabilities of bases
-			//(node will be BubbleUp prevNode and node.Up will be the node being operated on)
-			BubbleUp(node.Up, node, scrap) //node becomes PrevNode and scrap is set to one value of prevNode.Stored in BubbleUp
-			ans[i] = root.Scrap            //root.Stored has previously assigned values (SetInternalState), you want to use whatever is returned by BubbleUp instead
-		} else if node.Up == nil {
-			ans[i] = root.Stored[i]
+	for n := 0; n < len(g.Nodes); n++ {
```

Comment:
What happens if your graph has multiple starts/ends or is discontiguous?

Reply:
Ok, thats fair. The circumstances would like a genome graph with multiple chromosomes. There is a BreakDiscontiguousGraph function, but if this is mainly for small scale stuff it is fine as is.

Reply:
ah, ok i see, no i dont think we have to worry about that with this function, and I'm aware of the BreakDiscontiguous func too

Reply:
For a while I have been wondering if our graph data struct (along with a slice of all the nodes) should have a list of all the "start" nodes (maybe the "end" nodes too).

Reply:
I think thats a great idea, i think no matter what you're doing with the graph that info is useful enough to add it to the existing struct